std/ui/components/Video.tsx

```diff
@@ -0,0 +1,34 @@
+/**
+ * TODO: Implement video preload with link[rel="preload"] tags once
+ * browsers support it. More info at: https://stackoverflow.com/a/68368601
+ */
+import { forwardRef } from "preact/compat";
+import type { JSX } from "preact";
+
+import { getSrcSet } from "./Image.tsx";
+
+type Props =
+  & Omit<JSX.IntrinsicElements["video"], "width" | "height" | "preload">
+  & {
+    width: number;
+    height: number;
+    src: string;
+  };
+
+const Video = forwardRef<HTMLVideoElement, Props>((props, ref) => {
+  const { loading = "lazy" } = props;
+  const srcSet = getSrcSet(props.src, props.width, props.height);
+
+  return (
```

Comment:
wdyt about preloading with a link tag as fetch?

Reply:
looks like we need to do some wild stuff: https://web.dev/fast-playback-with-preload/#preload-the-first-segment

Let's wait for the link[rel=preload][as=video] support first